packages/puppeteer-core/src/bidi/Page.ts

```diff
@@ -219,6 +220,26 @@ export class BidiPage extends Page {
     await this._client().send('Page.setBypassCSP', {enabled});
   }
 
+  override async queryObjects<Prototype>(
+    prototypeHandle: BidiJSHandle<Prototype>
+  ): Promise<BidiJSHandle<Prototype[]>> {
+    assert(!prototypeHandle.disposed, 'Prototype JSHandle is disposed!');
+    assert(
+      prototypeHandle.id,
+      'Prototype JSHandle must not be referencing primitive value'
+    );
+    const response = await this.mainFrame().client.send(
```

Comment:
note that we don't worry about mparch here because the state is connected to the main frame's realm